crates/fluvio-spu/src/services/public/tests/produce.rs

```diff
@@ -425,7 +442,9 @@ async fn test_produce_metrics() {
     spu_config.log.base_dir = test_path;
     let ctx = GlobalContext::new_shared_context(spu_config);
 
-    let server_end_event = create_public_server(addr.to_owned(), ctx.clone()).run();
+    let auth_global_ctx =
+        SpuAuthGlobalContext::new(ctx.clone(), Arc::new(RootAuthorization::new()));
```

Comment:
this is repeated. can we try to create util so all tests can just same pattern?

Reply:
Sure! I created a `create_public_server_with_root_auth` function for most test cases.